source/mothership-design.md

```diff
@@ -0,0 +1,771 @@
+| `SYST`, `KILL`  | None                  | Stops processing of further       |
+|                 |                       | messages and packets, and shuts   |
+|                 |                       | down the Mothership process as    |
+|                 |                       | *quickly* as possible.            |
++-----------------+-----------------------+-----------------------------------+
+| `APP`,  `SPEC`  | 0. `std::string`      | Defines that an application on    |
+|                 |    `appName`          | the receiving Mothership process  |
+|                 | 1. `uint32_t`         | must have received `distCount`    |
+|                 |    `distCount`        | unique distribution (`APP`,       |
+|                 | 2. `uint8_t`          | `DIST` and `APP`, `SUPD`)         |
+|                 |    `appNumber`        | messages in order to be fully     |
+|                 |                       | defined.                          |
++-----------------+-----------------------+-----------------------------------+
+| `APP`,  `DIST`  | 0. `std::string`      | Defines the properties for a      |
+|                 |    `appName`          | given core for a given            |
+|                 | 1. `std::string`      | application on this Mothership    |
+|                 |    `codePath`         | process.                          |
+|                 | 2. `std::string`      |                                   |
+|                 |    `dataPath`         |                                   |
+|                 | 3. `uint32_t`         |                                   |
+|                 |    `coreAddr`         |                                   |
+|                 | 4. `std::vector<`     |                                   |
+|                 |    `uint32_t>`        |                                   |
+|                 |    `threadsExpected`  |                                   |
++-----------------+-----------------------+-----------------------------------+
+| `APP`,  `SUPD`  | 0. `std::string`      | Defines the properties for the    |
+|                 |    `appName`          | supervisor for a given            |
+|                 | 1. `std::string`      | application on this Mothership.   |
+|                 |    `soPath`           |                                   |
++-----------------+-----------------------+-----------------------------------+
+| `CMND`,  `RECL` | 0. `std::string`      | Removes information for an        |
+|                 |    `appName`          | application, by name, from the    |
+|                 |                       | Mothership. Does nothing on a     |
+|                 |                       | running application (it must be   |
+|                 |                       | stopped first).                   |
++-----------------+-----------------------+-----------------------------------+
+| `CMND`, `INIT`  | 0. `std::string`      | Takes a fully-defined             |
```

Comment:
> Is this message (and the next two) essentially queued indefinitely?

The message is queued until it is consumed (which is what I think you mean). Your example is correct.

The behaviour for all `CMND` inputs is that the command is staged, regardless of whether or not the Mothership process otherwise "knows" about the application.

I've added a sentence to the top paragraph of the MPI Command and Control section to explain this.

Reply:
:+1: